addon/components/polaris-button.js

```diff
@@ -322,107 +318,122 @@ export default Component.extend({
+      'fullWidth',
+      'isIconOnly'
+    );
+    let classes = ['Polaris-Button'];
+
+    if (isPresent(externalClasses)) {
+      classes.push(externalClasses);
+    }
 
-      if (outline) {
-        classes.push('Polaris-Button--outline');
-      }
+    if (primary) {
+      classes.push('Polaris-Button--primary');
+    }
 
-      if (destructive) {
-        classes.push('Polaris-Button--destructive');
-      }
+    if (outline) {
+      classes.push('Polaris-Button--outline');
+    }
 
-      if (isDisabled) {
-        classes.push('Polaris-Button--disabled');
-      }
+    if (destructive) {
+      classes.push('Polaris-Button--destructive');
+    }
 
-      if (loading) {
-        classes.push('Polaris-Button--loading');
-      }
+    if (isDisabled) {
+      classes.push('Polaris-Button--disabled');
+    }
 
-      if (plain) {
-        classes.push('Polaris-Button--plain');
-      }
+    if (loading) {
+      classes.push('Polaris-Button--loading');
+    }
 
-      if (monochrome) {
-        classes.push('Polaris-Button--monochrome');
-      }
+    if (plain) {
+      classes.push('Polaris-Button--plain');
+    }
 
-      if (size && size !== DEFAULT_SIZE) {
-        size = SIZES[size] || null;
-        if (size) {
-          classes.push(`Polaris-Button--size${classify(size)}`);
-        }
-      }
+    if (monochrome) {
+      classes.push('Polaris-Button--monochrome');
+    }
 
-      if (fullWidth) {
-        classes.push('Polaris-Button--fullWidth');
+    if (size && size !== DEFAULT_SIZE) {
+      size = SIZES[size] || null;
+      if (size) {
+        classes.push(`Polaris-Button--size${classify(size)}`);
       }
+    }
 
-      if (isIconOnly) {
-        classes.push('Polaris-Button--iconOnly');
-      }
+    if (fullWidth) {
+      classes.push('Polaris-Button--fullWidth');
+    }
 
-      return classes.join(' ');
+    if (isIconOnly) {
+      classes.push('Polaris-Button--iconOnly');
     }
-  ).readOnly(),
 
-  type: computed('submit', function() {
+    return classes.join(' ');
+  }
+
+  @(computed('submit').readOnly())
+  get type() {
     return this.get('submit') === true ? 'submit' : 'button';
-  }).readOnly(),
+  }
 
-  ariaExpandedValue: computed('ariaExpanded', function() {
+  @(computed('ariaExpanded').readOnly())
+  get ariaExpandedValue() {
     let ariaExpanded = this.get('ariaExpanded');
     return isPresent(ariaExpanded) ? String(ariaExpanded) : null;
-  }).readOnly(),
+  }
 
-  ariaPressedValue: computed('ariaPressed', function() {
+  @(computed('ariaPressed').readOnly())
+  get ariaPressedValue() {
     let ariaPressed = this.get('ariaPressed');
     return isPresent(ariaPressed) ? String(ariaPressed) : null;
-  }).readOnly(),
-
-  actions: {
-    /**
-     * Helper to invoke passed-in actions without passing the
-     * button element's event object as the first parameter.
-     */
-    invokeMouseAction(actionName) {
-      return this.get(actionName)();
-    },
-  },
-});
+  }
+
+  init() {
```

Comment:
init should be replaced with constructor

Reply:
for components? do you have a source for this?
I was under the impression that it's recommended to still use `init` if extending from an `EmberObject` based class

Reply:
my bad. you're right 🙂I overlooked the fact that this is still an ember component and not a glimmer component.

Reply:
no worries...thinking of actually switching to a glimmer component while here